template.md

```diff
@@ -0,0 +1,305 @@
+- If an optional metadata element does not have a value, comment out the element with a # (do not leave it blank or use "na"); if you are adding a value to an element that was commented out, be sure to remove the #.
+- Colons in a value (for example, a title) break the metadata parser. In their place, use the HTML encoding for a colon of `&#58;` (for example, `"title: Writing .NET Core console apps&#58 An advanced step-by-step guide"`).
+- **title**: This title will appear in search engine results. The title should end with a pipe (|) followed by the name of the service (for example, see above). The title doesn't need (and probably shouldn't) be identical to the title in your H1 heading. It should be roughly 65 characters (including | SERVICE NAME)
+- **author**, **manager**, **reviewer**: The author field should contain the **GitHub username** of the author, not their alias.  The "manager" and "reviewer" fields, on the other hand, should contain aliases. ms.reviewer specifies the name of the PM associated with the article or service.
+- **ms.assetid**: This is the GUID of the article from CAPS. When creating a new Markdown file, get a GUID from [https://www.guidgenerator.com](https://www.guidgenerator.com). 
+
+## Basic Markdown, GFM, and special characters
+
+All basic and GitHub-flavored Markdown is supported. For more information on these, see:
+
+- [Baseline Markdown syntax](https://daringfireball.net/projects/markdown/syntax)
+- [GitHub-flavored Markdown (GFM) documentation](https://guides.github.com/features/mastering-markdown)
+
+Markdown uses special characters such as \*, \`, and \# for formatting. If you wish to include one of these characters in your content, you must do one of two things:
+
+- Put a backslash before the special character to "escape" it (for example, \\\* for a \*)
+- Use the [HTML entity code](http://www.ascii.cl/htmlcodes.htm) for the character (for example, \&\#42\; for a &#42;).
+
+## File name
+
+File names use the following rules:
+* Contain only lowercase letters, numbers, and hyphens.
+* No spaces or punctuation characters. Use the hyphens to separate words and numbers in the file name.
+* No more than 80 characters - this is a publishing system limit.
+* Use action verbs that are specific, such as develop, buy, build, troubleshoot. No -ing words.
+* No small words - don't include a, and, the, in, or, etc.
+* Must be in Markdown and use the .md file extension.
+
+## Headings
+
+Headings should be done using atx-style, that is, use 1-6 hash characters (#) at the start of the line to indicate a heading, corresponding to HTML headings levels H1 through H6. Examples of first- and second-level headers are used above. 
+
+There **must** be only one first-level heading (H1) in your topic, which will be displayed as the on-page title.  
+
+Second-level headings will generate the on-page TOC that appears in the "In this article" section underneath the on-page title.
```

Comment:
Since `#` is relatively common character in headings, is is worth pointing out that it requires special handling when it's the last character of the heading? E.g. `# Learn C# #`.